src/renderer/containers/main/multi_editor.ts

```diff
@@ -48,8 +48,36 @@ class MultiEditor extends Container<MultiEditorState, MainCTX> {
 
   }
 
+  _deduplicate ( arr: any[] = [] ) {
+    Array.from( new Set( arr ) )
+  }
+
   /* API */
 
+  toggleNoteRange = ( note: NoteObj ) => {
+
+    const selectedNotes = this.getNotes(),
+          notes = this.ctx.search.getNotes(),
+          anchorNoteIndex = notes.indexOf(selectedNotes[selectedNotes.length - 1]),
+          markerNoteIndex = notes.indexOf(note);
+
+    let startIndex, endIndex;
+
+    if (anchorNoteIndex > markerNoteIndex) {
+
+      startIndex = markerNoteIndex, endIndex = anchorNoteIndex;
+
+    } else {
+
+      startIndex = anchorNoteIndex, endIndex = markerNoteIndex;
+
+    }
+
+    let notesNext = [...selectedNotes, ...notes.slice(startIndex, endIndex + 1)];
```

Comment:
It looks like the previously selected notes are always preserved, this is not the behavior you can see [here](http://demo.getsvelto.com/widgets/selectable.html), notes in the range are instead toggled.